src/proof/verify.rs

```diff
@@ -107,6 +96,31 @@ impl Deref for NodeDecodingResult {
     }
 }
 
+#[inline]
+fn process_trie_node(
+    node: TrieNode,
+    walked_path: &mut Nibbles,
+    key: &Nibbles,
+) -> Result<Option<NodeDecodingResult>, ProofVerificationError> {
+    let node = match node {
+        TrieNode::Branch(branch) => process_branch(branch, walked_path, key)?,
+        TrieNode::Extension(extension) => {
+            walked_path.extend_from_slice(&extension.key);
+            if extension.child.len() == B256::len_bytes() + 1 {
```

Comment:
yeah can we convert this into a `is_hash` function to make this obvious why we're checking this?